lib/ibanity/util.rb

```diff
@@ -1,9 +1,16 @@
 module Ibanity
   module Util
-    def self.underscorize_hash(attributes)
-      attributes.keys.reduce({}) do |result, key|
-        result[underscore(key)] = attributes[key]
-        result
+    def self.underscorize(obj)
+      case obj
+      when Array
+        obj.map { |e| underscorize(e) }
+      when Hash
+        obj.keys.reduce({}) do |result, key|
+          result[underscore(key)] = underscorize(obj[key])
```

Comment:
This is a good improvement, however it is a breaking change... We should definitely add it to the changelog and also bump the major version.

Reply:
I agree